src/Rules/Library/NameIsEmptyButElementIsNotKeyboardFocusable.cs

```diff
@@ -21,13 +22,14 @@ public NameIsEmptyButElementIsNotKeyboardFocusable()
             this.Info.HowToFix = HowToFix.NameEmptyButElementNotKeyboardFocusable;
             this.Info.Standard = A11yCriteriaId.ObjectInformation;
             this.Info.PropertyID = PropertyType.UIA_NamePropertyId;
+            this.Info.ErrorCode = EvaluationCode.Open;
         }
 
-        public override EvaluationCode Evaluate(IA11yElement e)
+        public override bool PassesTest(IA11yElement e)
         {
-            if (e == null) return EvaluationCode.RuleExecutionError;
+            if (e == null) throw new ArgumentNullException(nameof(e));
```

Comment:
This Exception makes sense, but there are lots of other rules that will merrily throw a NulReferenceException in this case (for example, NameIsInformative (chosen just because it's the next in the review) has no guard. It feels like the guard ought to be consistent in all rules (could be a separate PR)

Reply:
Agreed.